net/base/directory_lister.h

```diff
@@ -21,11 +21,119 @@ class TaskRunner;
 
 namespace net {
 
-// This class provides an API for asynchronously listing the contents of a
-// directory on the filesystem.  It runs a task on a background thread, and
-// enumerates all files in the specified directory on that thread.  Destroying
-// the lister cancels the list operation.  The DirectoryLister must only be
-// used on a thread with a MessageLoop.
+#if defined(STARBOARD)
+// TODO b/296715826 Fix stubbed out class.
+class NET_EXPORT DirectoryLister  {
+ public:
+  // Represents one file found.
+  struct DirectoryListerData {
+    base::FileEnumerator::FileInfo info;
+    base::FilePath path;
+    base::FilePath absolute_path;
+  };
+
+  // Implement this class to receive directory entries.
+  class DirectoryListerDelegate {
+   public:
+    // Called for each file found by the lister.
+    virtual void OnListFile(const DirectoryListerData& data) = 0;
+
+    // Called when the listing is complete.
+    virtual void OnListDone(int error) = 0;
+
+   protected:
+    virtual ~DirectoryListerDelegate() {}
+  };
+
+  // Listing options
+  // ALPHA_DIRS_FIRST is the default listing type:
+  //   directories first in name order, then files by name order
+  // Listing is recursive only if listing type is NO_SORT_RECURSIVE.
+  // TODO(mmenke): Improve testing.
+  enum ListingType {
+    NO_SORT,
+    NO_SORT_RECURSIVE,
+    ALPHA_DIRS_FIRST,
+  };
+
+  DirectoryLister(const base::FilePath& dir,
+                  DirectoryListerDelegate* delegate){};
+
+  DirectoryLister(const base::FilePath& dir,
+                  ListingType type,
+                  DirectoryListerDelegate* delegate){};
+
+  // Will invoke Cancel().
+  ~DirectoryLister(){};
+
+  // Call this method to start the asynchronous directory enumeration.
+  void Start() {};
+
+  // Call this method to asynchronously stop directory enumeration.  The
+  // delegate will not be called back.
+  void Cancel() {};
+
+ private:
+  typedef std::vector<DirectoryListerData> DirectoryList;
```

Comment:
I wouldn't think any private members are actually needed - you just need a stubbed out public interface. Try getting rid of everything within `private:`